diff --git a/common/MakefileCommonGCC.mk b/common/MakefileCommonGCC.mk
index 74666637..1623f28a 100644
--- a/common/MakefileCommonGCC.mk
+++ b/common/MakefileCommonGCC.mk
@@ -521,6 +521,8 @@ IMAGE_LISTING  := $(IMAGE_BUILD_DIR)/$(IMAGE_NAME).lst
 IMAGE_SREFILE  := $(IMAGE_BUILD_DIR)/$(IMAGE_NAME).sre
 IMAGE_SYMFILE  := $(IMAGE_BUILD_DIR)/$(IMAGE_NAME).sym
 IMAGE_MAP_FILE := $(IMAGE_BUILD_DIR)/$(IMAGE_NAME).map
+IMAGE_SIZECODE := $(IMAGE_BUILD_DIR)/$(IMAGE_NAME)
+
 
 image: $(IMAGE_HEXFILE)
 $(IMAGE_HEXFILE): $(IMAGE_FILE)
@@ -539,8 +541,9 @@ $(IMAGE_LISTING): $(IMAGE_FILE)
 	@$(ECHO) $(___TARGET_SEPARATOR_LINE___)
 	@$(ECHO) Generating listing file $(IMAGE_LISTING) from "$(IMAGE_FILE)"
 	@$(LISTING_TOOL_CMD)
-
-image_size: $(IMAGE_FILE)
+	
+image: $(IMAGE_SIZECODE)
+$(IMAGE_SIZECODE): $(IMAGE_FILE) 
 	@$(ECHO) $(___TARGET_SEPARATOR_LINE___)
 	@$(ECHO) Calculating size from "$(IMAGE_FILE)"
 	@$(SIZE_TOOL_CMD)
diff --git a/projects/patch_DR_HSC_Infineon/.cproject b/projects/patch_DR_HSC_Infineon/.cproject
index 0e567606..0abdb14c 100644
--- a/projects/patch_DR_HSC_Infineon/.cproject
+++ b/projects/patch_DR_HSC_Infineon/.cproject
@@ -1,6 +1,42 @@
 <?xml version="1.0" encoding="UTF-8" standalone="no"?>
 <?fileVersion 4.0.0?><cproject storage_type_id="org.eclipse.cdt.core.XmlProjectDescriptionStorage">
 	<storageModule moduleId="org.eclipse.cdt.core.settings">
+		<cconfiguration id="0.1799369606.1812942932">
+			<storageModule buildSystemId="org.eclipse.cdt.managedbuilder.core.configurationDataProvider" id="0.1799369606.1812942932" moduleId="org.eclipse.cdt.core.settings" name="Windows">
+				<externalSettings/>
+				<extensions>
+					<extension id="org.eclipse.cdt.core.VCErrorParser" point="org.eclipse.cdt.core.ErrorParser"/>
+					<extension id="org.eclipse.cdt.core.GmakeErrorParser" point="org.eclipse.cdt.core.ErrorParser"/>
+					<extension id="org.eclipse.cdt.core.CWDLocator" point="org.eclipse.cdt.core.ErrorParser"/>
+					<extension id="org.eclipse.cdt.core.GCCErrorParser" point="org.eclipse.cdt.core.ErrorParser"/>
+					<extension id="org.eclipse.cdt.core.GASErrorParser" point="org.eclipse.cdt.core.ErrorParser"/>
+					<extension id="org.eclipse.cdt.core.GLDErrorParser" point="org.eclipse.cdt.core.ErrorParser"/>
+					<extension id="org.eclipse.cdt.ui.PC-Lint" point="org.eclipse.cdt.core.ErrorParser"/>
+					<extension id="org.eclipse.cdt.ui.Keil" point="org.eclipse.cdt.core.ErrorParser"/>
+				</extensions>
+			</storageModule>
+			<storageModule moduleId="cdtBuildSystem" version="4.0.0">
+				<configuration artifactName="B13.OPUS.patch" buildProperties="" description="" errorParsers="org.eclipse.cdt.core.GASErrorParser;org.eclipse.cdt.core.GmakeErrorParser;org.eclipse.cdt.core.GLDErrorParser;org.eclipse.cdt.core.VCErrorParser;org.eclipse.cdt.core.CWDLocator;org.eclipse.cdt.core.GCCErrorParser;org.eclipse.cdt.ui.Keil;org.eclipse.cdt.ui.PC-Lint" id="0.1799369606.1812942932" name="Windows" optionalBuildProperties="" parent="org.eclipse.cdt.build.core.prefbase.cfg">
+					<folderInfo id="0.1799369606.1812942932." name="/" resourcePath="">
+						<toolChain errorParsers="" id="org.eclipse.cdt.build.core.prefbase.toolchain.104458540" name="No ToolChain" resourceTypeBasedDiscovery="false" superClass="org.eclipse.cdt.build.core.prefbase.toolchain">
+							<targetPlatform id="org.eclipse.cdt.build.core.prefbase.toolchain.104458540.837491438" name=""/>
+							<builder command="../../tools/GNUmake/make" errorParsers="org.eclipse.cdt.core.GmakeErrorParser;org.eclipse.cdt.core.CWDLocator" id="org.eclipse.cdt.build.core.settings.default.builder.1335284835" keepEnvironmentInBuildfile="false" managedBuildOn="false" name="Gnu Make Builder" superClass="org.eclipse.cdt.build.core.settings.default.builder"/>
+							<tool errorParsers="org.eclipse.cdt.core.GASErrorParser;org.eclipse.cdt.core.GLDErrorParser;org.eclipse.cdt.core.VCErrorParser;org.eclipse.cdt.core.GCCErrorParser" id="org.eclipse.cdt.build.core.settings.holder.libs.519577151" name="holder for library settings" superClass="org.eclipse.cdt.build.core.settings.holder.libs"/>
+							<tool errorParsers="org.eclipse.cdt.core.GASErrorParser;org.eclipse.cdt.core.GLDErrorParser;org.eclipse.cdt.core.VCErrorParser;org.eclipse.cdt.core.GCCErrorParser" id="org.eclipse.cdt.build.core.settings.holder.885808431" name="Assembly" superClass="org.eclipse.cdt.build.core.settings.holder">
+								<inputType id="org.eclipse.cdt.build.core.settings.holder.inType.247336633" languageId="org.eclipse.cdt.core.assembly" languageName="Assembly" sourceContentType="org.eclipse.cdt.core.asmSource" superClass="org.eclipse.cdt.build.core.settings.holder.inType"/>
+							</tool>
+							<tool errorParsers="org.eclipse.cdt.core.GASErrorParser;org.eclipse.cdt.core.GLDErrorParser;org.eclipse.cdt.core.VCErrorParser;org.eclipse.cdt.core.GCCErrorParser" id="org.eclipse.cdt.build.core.settings.holder.1772217517" name="GNU C++" superClass="org.eclipse.cdt.build.core.settings.holder">
+								<inputType id="org.eclipse.cdt.build.core.settings.holder.inType.1969018278" languageId="org.eclipse.cdt.core.g++" languageName="GNU C++" sourceContentType="org.eclipse.cdt.core.cxxSource,org.eclipse.cdt.core.cxxHeader" superClass="org.eclipse.cdt.build.core.settings.holder.inType"/>
+							</tool>
+							<tool errorParsers="org.eclipse.cdt.core.GASErrorParser;org.eclipse.cdt.core.GLDErrorParser;org.eclipse.cdt.core.VCErrorParser;org.eclipse.cdt.core.GCCErrorParser" id="org.eclipse.cdt.build.core.settings.holder.955798280" name="GNU C" superClass="org.eclipse.cdt.build.core.settings.holder">
+								<inputType id="org.eclipse.cdt.build.core.settings.holder.inType.1086976418" languageId="org.eclipse.cdt.core.gcc" languageName="GNU C" sourceContentType="org.eclipse.cdt.core.cSource,org.eclipse.cdt.core.cHeader" superClass="org.eclipse.cdt.build.core.settings.holder.inType"/>
+							</tool>
+						</toolChain>
+					</folderInfo>
+				</configuration>
+			</storageModule>
+			<storageModule moduleId="org.eclipse.cdt.core.externalSettings"/>
+		</cconfiguration>
 		<cconfiguration id="0.1799369606">
 			<storageModule buildSystemId="org.eclipse.cdt.managedbuilder.core.configurationDataProvider" id="0.1799369606" moduleId="org.eclipse.cdt.core.settings" name="Linux">
 				<externalSettings/>
@@ -16,26 +52,26 @@
 				</extensions>
 			</storageModule>
 			<storageModule moduleId="cdtBuildSystem" version="4.0.0">
-				<configuration artifactName="B13.OPUS.patch" buildProperties="" description="" errorParsers="org.eclipse.cdt.core.GASErrorParser;org.eclipse.cdt.core.GmakeErrorParser;org.eclipse.cdt.core.GLDErrorParser;org.eclipse.cdt.core.VCErrorParser;org.eclipse.cdt.core.CWDLocator;org.eclipse.cdt.core.GCCErrorParser;org.eclipse.cdt.ui.Keil;org.eclipse.cdt.ui.PC-Lint" id="0.1799369606" name="Linux" optionalBuildProperties="" parent="org.eclipse.cdt.build.core.prefbase.cfg">
+				<configuration artifactName="B13.OPUS.patch" buildProperties="" description="" errorParsers="org.eclipse.cdt.core.GASErrorParser;org.eclipse.cdt.core.GmakeErrorParser;org.eclipse.cdt.core.GLDErrorParser;org.eclipse.cdt.core.VCErrorParser;org.eclipse.cdt.core.CWDLocator;org.eclipse.cdt.core.GCCErrorParser;org.eclipse.cdt.ui.Keil;org.eclipse.cdt.ui.PC-Lint" id="0.1799369606" name="Linux" parent="org.eclipse.cdt.build.core.prefbase.cfg">
 					<folderInfo id="0.1799369606." name="/" resourcePath="">
 						<toolChain errorParsers="" id="org.eclipse.cdt.build.core.prefbase.toolchain.119278819" name="No ToolChain" resourceTypeBasedDiscovery="false" superClass="org.eclipse.cdt.build.core.prefbase.toolchain">
 							<targetPlatform id="org.eclipse.cdt.build.core.prefbase.toolchain.119278819.1697297017" name=""/>
 							<builder errorParsers="org.eclipse.cdt.core.GmakeErrorParser;org.eclipse.cdt.core.CWDLocator" id="org.eclipse.cdt.build.core.settings.default.builder.1693845684" keepEnvironmentInBuildfile="false" managedBuildOn="false" name="Gnu Make Builder" superClass="org.eclipse.cdt.build.core.settings.default.builder"/>
 							<tool errorParsers="org.eclipse.cdt.core.GASErrorParser;org.eclipse.cdt.core.GLDErrorParser;org.eclipse.cdt.core.VCErrorParser;org.eclipse.cdt.core.GCCErrorParser" id="org.eclipse.cdt.build.core.settings.holder.libs.1492581623" name="holder for library settings" superClass="org.eclipse.cdt.build.core.settings.holder.libs"/>
 							<tool errorParsers="org.eclipse.cdt.core.GASErrorParser;org.eclipse.cdt.core.GLDErrorParser;org.eclipse.cdt.core.VCErrorParser;org.eclipse.cdt.core.GCCErrorParser" id="org.eclipse.cdt.build.core.settings.holder.569815426" name="Assembly" superClass="org.eclipse.cdt.build.core.settings.holder">
-								<option IS_BUILTIN_EMPTY="false" IS_VALUE_EMPTY="false" id="org.eclipse.cdt.build.core.settings.holder.symbols.704023839" name="Symbols" superClass="org.eclipse.cdt.build.core.settings.holder.symbols" valueType="definedSymbols">
+								<option id="org.eclipse.cdt.build.core.settings.holder.symbols.704023839" name="Symbols" superClass="org.eclipse.cdt.build.core.settings.holder.symbols" valueType="definedSymbols">
 									<listOptionValue builtIn="false" value="__GNUC__=1"/>
 								</option>
 								<inputType id="org.eclipse.cdt.build.core.settings.holder.inType.259653600" languageId="org.eclipse.cdt.core.assembly" languageName="Assembly" sourceContentType="org.eclipse.cdt.core.asmSource" superClass="org.eclipse.cdt.build.core.settings.holder.inType"/>
 							</tool>
 							<tool errorParsers="org.eclipse.cdt.core.GASErrorParser;org.eclipse.cdt.core.GLDErrorParser;org.eclipse.cdt.core.VCErrorParser;org.eclipse.cdt.core.GCCErrorParser" id="org.eclipse.cdt.build.core.settings.holder.2030369760" name="GNU C++" superClass="org.eclipse.cdt.build.core.settings.holder">
-								<option IS_BUILTIN_EMPTY="false" IS_VALUE_EMPTY="false" id="org.eclipse.cdt.build.core.settings.holder.symbols.1031907613" name="Symbols" superClass="org.eclipse.cdt.build.core.settings.holder.symbols" valueType="definedSymbols">
+								<option id="org.eclipse.cdt.build.core.settings.holder.symbols.1031907613" name="Symbols" superClass="org.eclipse.cdt.build.core.settings.holder.symbols" valueType="definedSymbols">
 									<listOptionValue builtIn="false" value="__GNUC__=1"/>
 								</option>
 								<inputType id="org.eclipse.cdt.build.core.settings.holder.inType.1290705135" languageId="org.eclipse.cdt.core.g++" languageName="GNU C++" sourceContentType="org.eclipse.cdt.core.cxxSource,org.eclipse.cdt.core.cxxHeader" superClass="org.eclipse.cdt.build.core.settings.holder.inType"/>
 							</tool>
 							<tool errorParsers="org.eclipse.cdt.core.GASErrorParser;org.eclipse.cdt.core.GLDErrorParser;org.eclipse.cdt.core.VCErrorParser;org.eclipse.cdt.core.GCCErrorParser" id="org.eclipse.cdt.build.core.settings.holder.1925120734" name="GNU C" superClass="org.eclipse.cdt.build.core.settings.holder">
-								<option IS_BUILTIN_EMPTY="false" IS_VALUE_EMPTY="false" id="org.eclipse.cdt.build.core.settings.holder.symbols.173513511" name="Symbols" superClass="org.eclipse.cdt.build.core.settings.holder.symbols" valueType="definedSymbols">
+								<option id="org.eclipse.cdt.build.core.settings.holder.symbols.173513511" name="Symbols" superClass="org.eclipse.cdt.build.core.settings.holder.symbols" valueType="definedSymbols">
 									<listOptionValue builtIn="false" value="__GNUC__=1"/>
 								</option>
 								<inputType id="org.eclipse.cdt.build.core.settings.holder.inType.112680049" languageId="org.eclipse.cdt.core.gcc" languageName="GNU C" sourceContentType="org.eclipse.cdt.core.cSource,org.eclipse.cdt.core.cHeader" superClass="org.eclipse.cdt.build.core.settings.holder.inType"/>
@@ -48,42 +84,6 @@
 				<externalSettings containerId="ram_image;" factoryId="org.eclipse.cdt.core.cfg.export.settings.sipplier"/>
 			</storageModule>
 		</cconfiguration>
-		<cconfiguration id="0.1799369606.1812942932">
-			<storageModule buildSystemId="org.eclipse.cdt.managedbuilder.core.configurationDataProvider" id="0.1799369606.1812942932" moduleId="org.eclipse.cdt.core.settings" name="Windows">
-				<externalSettings/>
-				<extensions>
-					<extension id="org.eclipse.cdt.core.VCErrorParser" point="org.eclipse.cdt.core.ErrorParser"/>
-					<extension id="org.eclipse.cdt.core.GmakeErrorParser" point="org.eclipse.cdt.core.ErrorParser"/>
-					<extension id="org.eclipse.cdt.core.CWDLocator" point="org.eclipse.cdt.core.ErrorParser"/>
-					<extension id="org.eclipse.cdt.core.GCCErrorParser" point="org.eclipse.cdt.core.ErrorParser"/>
-					<extension id="org.eclipse.cdt.core.GASErrorParser" point="org.eclipse.cdt.core.ErrorParser"/>
-					<extension id="org.eclipse.cdt.core.GLDErrorParser" point="org.eclipse.cdt.core.ErrorParser"/>
-					<extension id="org.eclipse.cdt.ui.PC-Lint" point="org.eclipse.cdt.core.ErrorParser"/>
-					<extension id="org.eclipse.cdt.ui.Keil" point="org.eclipse.cdt.core.ErrorParser"/>
-				</extensions>
-			</storageModule>
-			<storageModule moduleId="cdtBuildSystem" version="4.0.0">
-				<configuration artifactName="B13.OPUS.patch" buildProperties="" description="" errorParsers="org.eclipse.cdt.core.GASErrorParser;org.eclipse.cdt.core.GmakeErrorParser;org.eclipse.cdt.core.GLDErrorParser;org.eclipse.cdt.core.VCErrorParser;org.eclipse.cdt.core.CWDLocator;org.eclipse.cdt.core.GCCErrorParser;org.eclipse.cdt.ui.Keil;org.eclipse.cdt.ui.PC-Lint" id="0.1799369606.1812942932" name="Windows" parent="org.eclipse.cdt.build.core.prefbase.cfg">
-					<folderInfo id="0.1799369606.1812942932." name="/" resourcePath="">
-						<toolChain errorParsers="" id="org.eclipse.cdt.build.core.prefbase.toolchain.104458540" name="No ToolChain" resourceTypeBasedDiscovery="false" superClass="org.eclipse.cdt.build.core.prefbase.toolchain">
-							<targetPlatform id="org.eclipse.cdt.build.core.prefbase.toolchain.104458540.837491438" name=""/>
-							<builder command="../../tools/GNUmake/make" errorParsers="org.eclipse.cdt.core.GmakeErrorParser;org.eclipse.cdt.core.CWDLocator" id="org.eclipse.cdt.build.core.settings.default.builder.1335284835" keepEnvironmentInBuildfile="false" managedBuildOn="false" name="Gnu Make Builder" superClass="org.eclipse.cdt.build.core.settings.default.builder"/>
-							<tool errorParsers="org.eclipse.cdt.core.GASErrorParser;org.eclipse.cdt.core.GLDErrorParser;org.eclipse.cdt.core.VCErrorParser;org.eclipse.cdt.core.GCCErrorParser" id="org.eclipse.cdt.build.core.settings.holder.libs.519577151" name="holder for library settings" superClass="org.eclipse.cdt.build.core.settings.holder.libs"/>
-							<tool errorParsers="org.eclipse.cdt.core.GASErrorParser;org.eclipse.cdt.core.GLDErrorParser;org.eclipse.cdt.core.VCErrorParser;org.eclipse.cdt.core.GCCErrorParser" id="org.eclipse.cdt.build.core.settings.holder.885808431" name="Assembly" superClass="org.eclipse.cdt.build.core.settings.holder">
-								<inputType id="org.eclipse.cdt.build.core.settings.holder.inType.247336633" languageId="org.eclipse.cdt.core.assembly" languageName="Assembly" sourceContentType="org.eclipse.cdt.core.asmSource" superClass="org.eclipse.cdt.build.core.settings.holder.inType"/>
-							</tool>
-							<tool errorParsers="org.eclipse.cdt.core.GASErrorParser;org.eclipse.cdt.core.GLDErrorParser;org.eclipse.cdt.core.VCErrorParser;org.eclipse.cdt.core.GCCErrorParser" id="org.eclipse.cdt.build.core.settings.holder.1772217517" name="GNU C++" superClass="org.eclipse.cdt.build.core.settings.holder">
-								<inputType id="org.eclipse.cdt.build.core.settings.holder.inType.1969018278" languageId="org.eclipse.cdt.core.g++" languageName="GNU C++" sourceContentType="org.eclipse.cdt.core.cxxSource,org.eclipse.cdt.core.cxxHeader" superClass="org.eclipse.cdt.build.core.settings.holder.inType"/>
-							</tool>
-							<tool errorParsers="org.eclipse.cdt.core.GASErrorParser;org.eclipse.cdt.core.GLDErrorParser;org.eclipse.cdt.core.VCErrorParser;org.eclipse.cdt.core.GCCErrorParser" id="org.eclipse.cdt.build.core.settings.holder.955798280" name="GNU C" superClass="org.eclipse.cdt.build.core.settings.holder">
-								<inputType id="org.eclipse.cdt.build.core.settings.holder.inType.1086976418" languageId="org.eclipse.cdt.core.gcc" languageName="GNU C" sourceContentType="org.eclipse.cdt.core.cSource,org.eclipse.cdt.core.cHeader" superClass="org.eclipse.cdt.build.core.settings.holder.inType"/>
-							</tool>
-						</toolChain>
-					</folderInfo>
-				</configuration>
-			</storageModule>
-			<storageModule moduleId="org.eclipse.cdt.core.externalSettings"/>
-		</cconfiguration>
 	</storageModule>
 	<storageModule moduleId="cdtBuildSystem" version="4.0.0">
 		<project id="B31.OPUS.patch.null.1119781443" name="B31.OPUS.patch"/>
@@ -140,14 +140,22 @@
 				<useDefaultCommand>true</useDefaultCommand>
 				<runAllBuilders>true</runAllBuilders>
 			</target>
-			<target name="pmbus DEVICE_TYPE=4x4" path="" targetID="org.eclipse.cdt.build.MakeTargetBuilder">
-				<buildCommand>make</buildCommand>
+			<target name="patching BUILD=DEFAULT" path="" targetID="org.eclipse.cdt.build.MakeTargetBuilder">
+				<buildCommand>../../tools/GNUmake/make</buildCommand>
+				<buildArguments/>
+				<buildTarget>patching BUILD=DEFAULT</buildTarget>
+				<stopOnError>true</stopOnError>
+				<useDefaultCommand>true</useDefaultCommand>
+				<runAllBuilders>true</runAllBuilders>
+			</target>
+			<target name="pmbus_default_array BUILD=YES" path="" targetID="org.eclipse.cdt.build.MakeTargetBuilder">
+				<buildCommand>../../tools/GNUmake/make</buildCommand>
 				<buildArguments/>
-				<buildTarget>pmbus DEVICE_TYPE=4x4</buildTarget>
+				<buildTarget>pmbus_default_array BUILD=YES</buildTarget>
 				<stopOnError>true</stopOnError>
 				<useDefaultCommand>true</useDefaultCommand>
 				<runAllBuilders>true</runAllBuilders>
 			</target>
 		</buildTargets>
 	</storageModule>
-</cproject>
+</cproject>
\ No newline at end of file
diff --git a/projects/patch_DR_HSC_Infineon/Makefile b/projects/patch_DR_HSC_Infineon/Makefile
index bf63ce1f..9d5b6ea4 100644
--- a/projects/patch_DR_HSC_Infineon/Makefile
+++ b/projects/patch_DR_HSC_Infineon/Makefile
@@ -9,8 +9,8 @@
 # is used as reference when generating the absolute path information 
 REPO_ROOT_DIR := ../..
 
-# Overwrite help as the first target so that this one is called when no argument is given to GNUmake.
-default: simvision
+# Overwrite help as the first target so that this one is called when no argument is given to GNUmake. simvision
+default: patch
 
 # Toolchain defines the tools used for compiling. 
 # Toolchain can be ARM_GCC, the GNU Compiler or ARMV5, the Keil-ARM compiler
@@ -59,18 +59,7 @@ CCOMPILER_DIRS_FILTER := \
     ../version/ \
 	$(REPO_ROOT_DIR)/common/system_header/gnu/sys/ \
 	$(REPO_ROOT_DIR)/common/system_header/c_keil/ \
-	$(REPO_ROOT_DIR)/common/modules/configurator/ \
-    $(REPO_ROOT_DIR)/common/modules/driver_common/ \
-    $(REPO_ROOT_DIR)/common/modules/job_agency/ \
-    $(REPO_ROOT_DIR)/common/modules/logger/ \
-    $(REPO_ROOT_DIR)/common/modules/oam/ \
-    $(REPO_ROOT_DIR)/common/modules/otp/ \
-    $(REPO_ROOT_DIR)/common/modules/patcher/ \
-    $(REPO_ROOT_DIR)/common/modules/pmbus/ \
-    $(REPO_ROOT_DIR)/common/modules/regulation/ \
-    $(REPO_ROOT_DIR)/common/modules/rtx/ \
-    $(REPO_ROOT_DIR)/common/modules/startup/ \
-    $(REPO_ROOT_DIR)/common/modules/user_app/
+	$(filter %/, $(wildcard $(REPO_ROOT_DIR)/common/modules/*/))
 
 CCOMPILER_DIRS := $(filter-out $(CCOMPILER_DIRS_FILTER), $(SOURCE_DIRS))
 
@@ -96,6 +85,7 @@ CCODEANALYZER_DIRS := $(filter-out $(CCODEANALYZER_DIRS_FILTER), $(CCOMPILER_DIR
 #PATCH_REF_IMAGE_NAMES := $(REPO_ROOT_DIR)/versions/Shasta.A11.ROM.v4_20180124_124011_rom_image
 #PATCH_REF_IMAGE_NAMES := $(REPO_ROOT_DIR)/projects/rom_image/build/image/image
 #PATCH_REF_IMAGE_NAMES := $(REPO_ROOT_DIR)/projects/rom_image/build/simvision/simvision
+#PATCH_REF_IMAGE_NAMES := $(REPO_ROOT_DIR)/versions/Shasta.A21.ROM.v9_20190309_173226_rom_image
 PATCH_REF_IMAGE_NAMES := $(REPO_ROOT_DIR)/versions/Shasta.A23.ROM.v12_20191118_093946_rom_image
 
 # Reference Image for the patch targets. Not relevant for the image target.
@@ -115,5 +105,20 @@ else
 # Plugin for the targets "patch".
 	include $(REPO_ROOT_DIR)/common/MakefileCommonARM.mk
 endif
+
+ifeq ($(BUILD), DEFAULT)
+	include $(REPO_ROOT_DIR)/common/MakefilePatchDefault.mk
+endif
+
+
+ifeq ($(BUILD), YES)
+	# Plugin for pmbus default table c code
+	include $(REPO_ROOT_DIR)/common/MakefilePmbusDefaultArrary.mk
+endif
+
 # Plugin for pmbus auto generation of c code
 include $(REPO_ROOT_DIR)/common/MakefilePmbus.mk
+
+
+
+
diff --git a/projects/patch_DR_HSC_Infineon/src/partial_patch.cfg b/projects/patch_DR_HSC_Infineon/src/partial_patch.cfg
index fb200028..0c4e7c1c 100644
--- a/projects/patch_DR_HSC_Infineon/src/partial_patch.cfg
+++ b/projects/patch_DR_HSC_Infineon/src/partial_patch.cfg
@@ -1,9 +1,9 @@
-[Patch]
-startaddress = 0x20063800
-endaddress = 0x20063fff
-minramsize = 0x700
-
-[Version]
-baseversion = 0x5DD2AD32
-newversion = 0x5971BBBB
-
+[Patch]
+startaddress = 0x20063800
+endaddress = 0x20063fff
+minramsize = 0x700
+
+[Version]
+baseversion = 0x5DD2AD32
+newversion = 0x6580F6F9
+
diff --git a/projects/patch_DR_HSC_Infineon/src/~$shasta_pmbus.xlsx b/projects/patch_DR_HSC_Infineon/src/~$shasta_pmbus.xlsx
deleted file mode 100644
index 96322cb5..00000000
Binary files a/projects/patch_DR_HSC_Infineon/src/~$shasta_pmbus.xlsx and /dev/null differ
diff --git a/tools/scripts/pmbus_xlsx2c.py b/tools/scripts/pmbus_xlsx2c.py
index 4fc47f87..273c58a2 100755
--- a/tools/scripts/pmbus_xlsx2c.py
+++ b/tools/scripts/pmbus_xlsx2c.py
@@ -49,7 +49,7 @@ def xlsx_parse(file_name):
     from openpyxl import load_workbook
     wb = load_workbook(filename=file_name, read_only=True)
     try:
-		snb = wb.get_sheet_names()
+		snb = wb.sheetnames
 		print snb
 
     except:
